e2e/setups.go

```diff
@@ -107,6 +108,42 @@ func AuthSetups() (ss []setup.Setup) {
 	return
 }
 
+func UpstreamSetups() []setup.Setup {
```

Comment:
I got a little confused with the setups naming. Part of the DefaultsSetups is also an Upstream. 

Maybe this is an `--insecure` flag test?

Reply:
I'd change that to a flag test.